### PR TITLE
Fix content generation workflow

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#24150|

## Description

Refactor job steps to avoid error on non existent branch

### CODE REVIEW

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 1]
```diff
@@ -98,7 +105,23 @@ jobs:
         with:
           submodules: recursive
           fetch-depth: 0
-          ref: ${{ matrix.wazuh_version }}
+
+      # Checkout to the tag. If it doesn't exist, continue with the branch
+      - name: Checkout to tag
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:110]: We can rename this and the "Checkout to branch" steps to better reflect the intent

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 2]
```diff
@@ -38,7 +38,6 @@ jobs:
         with:
           submodules: recursive
           fetch-depth: 0
-          ref: ${{ matrix.wazuh_version }}
 
       # Checkout to the tag. If it doesn't exist, continue with the branch
       - name: Checkout to tag
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:43]: We can rename this and the "Checkout to branch" steps to better reflect the intent

**Review by sebasfalcone — CHANGES_REQUESTED** [review]
Good solution

Just two minor details

File: .github/workflows/vulnerability-scanner-generate-database.yml (hunk 1 above)
Comment (MiguelazoDS) [reply .github/workflows/vulnerability-scanner-generate-database.yml]: done

File: .github/workflows/vulnerability-scanner-generate-database.yml (hunk 2 above)
Comment (MiguelazoDS) [reply .github/workflows/vulnerability-scanner-generate-database.yml]: done

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 3]
```diff
@@ -28,26 +28,33 @@ jobs:
...
+      - name: Checkout repository
         uses: actions/checkout@v3
         with:
           submodules: recursive
           fetch-depth: 0
-          ref: ${{ matrix.wazuh_version }}
 
       # Checkout to the tag. If it doesn't exist, continue with the branch
-      - name: Checkout to tag
+      - name: Checkout to ${{ matrix.wazuh_version }}
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:43]: +1: Good idea

**Review by sebasfalcone — APPROVED** [review]
LGTM!

**Pull request merged**